internal/api/core/submission.go

```diff
@@ -0,0 +1,73 @@
+package core
+
+import (
+	"context"
+
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type GradeableRequest interface {
+	GetContext() context.Context
+	GetAssignment() *model.Assignment
+	GetUserEmail() string
+	GetMessage() string
+	ToLogAttrs() []any
+}
```

Comment:
This seems like a lot for what is just an `core.APIRequestAssignmentContext` and message.
Why can't GradeToSubmissionResponse() just take those two instead of the interface?